optaplanner-examples/src/main/java/org/optaplanner/examples/conferencescheduling/domain/Talk.java

```diff
@@ -228,8 +230,10 @@ public int overlappingMutuallyExclusiveTalksTagCount(Talk other) {
         return (int) mutuallyExclusiveTalksTagSet.stream().filter(tag -> other.mutuallyExclusiveTalksTagSet.contains(tag)).count();
     }
 
-    public boolean hasPrerequisite(Talk talk) {
-        return prerequisiteTalkSet.contains(talk);
+    public int missingPrerequisiteCount() {
+        return (int) prerequisiteTalkSet.stream()
+                .filter(prerequisite -> prerequisite.getTimeslot() == null || timeslot.endsBefore(prerequisite.getTimeslot()))
```

Comment:
Good one that you fixed this. This affects the Construction Heuristics's intializingTrend.
Basically, it's better if by initializing more Talks (picking a room/timeslot for each one), the score only goes down, never up.